app/Services/GraphQL.php

```diff
@@ -15,14 +16,50 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client and return the data result.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @return array|null
      */
     public function query($query, $variables)
     {
-        return $this->client->query($query, $variables);
+        // Use try/catch to avoid any GraphQL related errors breaking the application.
+        try {
+            $response = $this->client->query($query, $variables);
+        } catch (\Exception $exception) {
+            Log::error('GraphQL request failed.', [
+                'query' => $query,
+                'variables' => $variables,
+                'exception' => $exception->getMessage(),
+            ]);
+
+            return null;
+        }
+
+        return $response ? $response->getData() : null;
```

Comment:
Instead of returning `null` here, would `[]` make sense to keep the return type consistent? That'd also help us avoid scenarios where we'd read a "key" from null (or have to use the `array_get` helper anywhere we call this function).

Reply:
I think that makes sense for consistency! (Though I feel like we'd still need to use `array_get` since otherwise, we might end up with an `undefined index` error?)

Reply:
If we always return an array from this method, then grabbing a nonexistent index should just be a warning (which sorta makes sense to me since that'd be unexpected behavior).